todo/src/reducers/todos.js

```diff
@@ -0,0 +1,13 @@
+import { NEW_TODO, TOGGLE_COMPLETE } from "../actions";
+
+export default (todos = [], action) => {
+  switch (action.type) {
+    case NEW_TODO:
+      return todos.concat(action.payload);
+    case TOGGLE_COMPLETE:
+      todos[action.payload].complete = !todos[action.payload].complete;
```

Comment:
This is mutating your original state. You are taking your state, called todos, and you are going to whatever index you get from action.payload, and you are overriding the complete property on the object at that index. Instead, we should map over the todos array, and where the index is equal to the index from the payload, we want to return a new object copied from the object at that index, with the complete property overridden. So take this line out
```suggestion
     
```